react-native

komidore64 opened the pull request.

komidore64 commented:
![aw yeah](https://adprice.fedorapeople.org/img/aw_yeah.png)

yithian commented:
`npm install` on my mac gives this output

```
npm WARN native-base@2.7.1 requires a peer of react@>=16.4.1 but none is installed. You must install peer dependencies yourself.
npm WARN native-base@2.7.1 requires a peer of react-native@>=0.56.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-native-maps@0.21.0 requires a peer of react-native@^0.51 || ^0.52 || ^0.53 || ^0.54 but none is installed. You must install peer dependencies yourself.

added 109 packages from 60 contributors, removed 81 packages, updated 79 packages and audited 37115 packages in 30.192s
found 7 vulnerabilities (4 low, 2 moderate, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```
`npm start` gives this output and fails to start:

```
> dustup@0.1.0 start /Users/yith/code/dustup
> ./scripts/startup_reqs.sh && react-native-scripts start

+ sudo sysctl -w fs.inotify.max_user_instances=1024
sysctl: unknown oid 'fs.inotify.max_user_instances'
+ sudo sysctl -w fs.inotify.max_user_watches=12288
sysctl: unknown oid 'fs.inotify.max_user_watches'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! dustup@0.1.0 start: `./scripts/startup_reqs.sh && react-native-scripts start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the dustup@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/yith/.npm/_logs/2018-07-15T20_14_53_958Z-debug.log
```

`npm audit fix` doesn't allow npm to start :(

komidore64 commented:
> npm install on my mac gives this output
> 
> npm WARN native-base@2.7.1 requires a peer of react@>=16.4.1 but none is installed. You must install peer dependencies yourself.
> npm WARN native-base@2.7.1 requires a peer of react-native@>=0.56.0 but none is installed. You must install peer dependencies yourself.
> npm WARN react-native-maps@0.21.0 requires a peer of react-native@^0.51 || ^0.52 || ^0.53 || ^0.54 but none is installed. You must install peer dependencies yourself.
> 
> added 109 packages from 60 contributors, removed 81 packages, updated 79 packages and audited 37115 packages in 30.192s
> found 7 vulnerabilities (4 low, 2 moderate, 1 high)
>   run `npm audit fix` to fix them, or `npm audit` for details

I get those warnings also. I'd like to take care of them at some point, but dustup still works for the time being.

komidore64 commented:
@yithian if you comment out the two `sysctl` lines in `scripts/startup_reqs.sh` does it work?

yithian commented:
@komidore64 nope, commenting those lines out makes `npm start` show this:

```
 npm start

> dustup@0.1.0 start /Users/yith/code/dustup
> ./scripts/startup_reqs.sh && react-native-scripts start

1:11:15 PM: Unable to start server
See https://git.io/v5vcn for more information, either install watchman or run the following snippet:
  sudo sysctl -w kern.maxfiles=5242880
  sudo sysctl -w kern.maxfilesperproc=524288

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! dustup@0.1.0 start: `./scripts/startup_reqs.sh && react-native-scripts start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the dustup@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/yith/.npm/_logs/2018-07-16T17_11_15_826Z-debug.log
```

komidore64 commented on data/games/guilty_gear_xrd_rev_2/movespecs/potemkin.json:
```diff
@@ -0,0 +1,125 @@
+      {
+        "name": "F.D.B.",
+        "inputs": [
+          "63214S"
```
@yithian is this correct? i was unable to find anything on dustloop about a held `S` variant for this move.

yithian replied:
Yeah, Pot can hold S for a bit to delay the flick. It's listed as a bullet point in F.D.B.

komidore64 replied:
Ah okay, cool. thank you

komidore64 replied:
dangit, i was looking at the revalator moveset, not rev2. :anguished:

komidore64 commented on components/MoveSectionItem.js:
```diff
@@ -11,6 +11,21 @@ import {
 import {SimpleMoveListItem} from 'dustup/components';
 
 export default class MoveSectionItem extends React.Component {
+  renderFollowUps(move) {
```
this could probably be extracted into its own component.

yithian commented:
`npm run exp` doesn't work for me but `npm start` does. all tests pass via `npm test`.

@komidore64 when do you think you'll hit a clean merging point?

komidore64 commented:
> npm run exp doesn't work for me but npm start does. all tests pass via npm test.
> 
> @komidore64 when do you think you'll hit a clean merging point?

I've been trying to keep it runnable and without error at any commit, so it should be mergeable whenever.

komidore64 commented:
@yithian pull latest, we'll make sure it doesn't crash and then merge if it's green?

komidore64 commented:
as discussed outside of this PR, i'm looking into how to have the project require specific versions of `node` and `npm`.

yithian commented:
we're a react-native project meow!

komidore64 commented:
![swig swoot](https://adprice.fedorapeople.org/img/swiggity_swooty.gif)
